### PR TITLE
Increase timeout period for get cluster state

### PR DESCRIPTION
**Pull request by aditya-jaishankar**
During gradient testing, it was noticed that there are cases where it takes longer than 300 seconds for a databricks cluster to spin down. This is incompatible with the default timeout period of 300 seconds and causes submission failures for successful job completions. The timeout is now increased to 600 seconds.

### CODE REVIEW

File: sync/_databricks.py [hunk 1]
```diff
@@ -1057,7 +1057,7 @@ def terminate_cluster(cluster_id: str) -> Response[dict]:
 
 
 def _wait_for_cluster_termination(
-    cluster_id: str, timeout_seconds=300, poll_seconds=10
+    cluster_id: str, timeout_seconds=600, poll_seconds=10
```
Comment (rmoneys) [inline sync/_databricks.py:1060]: Looks good. How about a logging statement just before the sleep though:
```
logging.info(f"Waiting for cluster {cluster_id} to terminate")
```

**Review by rmoneys — APPROVED** [review]

File: sync/_databricks.py [hunk 2]
```diff
@@ -1057,8 +1057,9 @@ def terminate_cluster(cluster_id: str) -> Response[dict]:
...
 ) -> Response[dict]:
+    logging.info(f"Waiting for cluster {cluster_id} to terminate")
```
Comment (aditya-jaishankar) [inline sync/_databricks.py:1062]: @rmoneys Nice idea, made the change.

**Review by gorskysd — DISMISSED** [review]
Small thing, but you might want to make sure this is updated to `main` and then increment the minor version in `sync/__init__.py`.

**Comment by aditya-jaishankar** [comment]
> Small thing, but you might want to make sure this is updated to `main` and then increment the minor version in `sync/__init__.py`.

Thanks for pointing this out @gorskysd. Done!

**Review by gorskysd — APPROVED** [review]
LGTM

**Comment by rmoneys** [comment]
Don't forget to always "Squash and merge"! Helps us make sense of the commits on main.

**Comment by rmoneys** [comment]
And delete the PR branch after merge too - there should be a button to do that on this page.

**Comment by aditya-jaishankar** [comment]
Whoops - will be better about this next time!

**Pull request merged**